qt/aqt/progress.py

```diff
@@ -185,11 +186,11 @@ def on_progress() -> None:
             assert self.mw
 
             progress = self.mw.backend.latest_progress()
-            if not (label := label_from_progress(progress)):
+            if not (update := progress_update(progress)):
                 return
             if dialog and dialog.wantCancel:
                 self.mw.backend.set_wants_abort()
-            self.update(label=label)
+            self.update(**asdict(update))
```

Comment:
I'd rather we explicitly pass in each arg, so we don't lose typechecking. We only need to worry about label, value and max. process is unused by update() and can be ignored, and I don't think maybeShow is used either.